pairing/bn254/twist.go

```diff
@@ -178,23 +178,25 @@ func (c *twistPoint) Mul(a *twistPoint, scalar *big.Int) {
 }
 
 func (c *twistPoint) MakeAffine() {
-	if c.z.IsOne() {
+	g := c.Clone()
+	if g.z.IsOne() {
 		return
-	} else if c.z.IsZero() {
-		c.x.SetZero()
-		c.y.SetOne()
-		c.t.SetZero()
+	} else if g.z.IsZero() {
+		g.x.SetZero()
+		g.y.SetOne()
+		g.t.SetZero()
 		return
 	}
 
-	zInv := (&gfP2{}).Invert(&c.z)
-	t := (&gfP2{}).Mul(&c.y, zInv)
+	zInv := (&gfP2{}).Invert(&g.z)
+	t := (&gfP2{}).Mul(&g.y, zInv)
 	zInv2 := (&gfP2{}).Square(zInv)
-	c.y.Mul(t, zInv2)
-	t.Mul(&c.x, zInv2)
-	c.x.Set(t)
-	c.z.SetOne()
-	c.t.SetOne()
+	g.y.Mul(t, zInv2)
+	t.Mul(&g.x, zInv2)
+	g.x.Set(t)
+	g.z.SetOne()
+	g.t.SetOne()
+	c.Set(g)
```

Comment:
Because of this Set here, I don't think using a Cloned `g` actually changes much, except it makes the _timing_ of a race more difficult to "hit" in prod/tests...
This function should definitively have a "not safe for concurrent use" in the comment.

Reply:
Yeah this bit is dicey. I added the "not safe for concurrent use" warning above the function.